src/core/channel-manager.ts

```diff
@@ -8,42 +9,46 @@ export type ChannelOptionsMerger = (options: Ably.ChannelOptions) => Ably.Channe
 export class ChannelManager {
```

Comment:
Do we really need this class? Previously, we needed it to handle complex logic.
But things stand as of now, simple logic inside `src/core/channel-manager.ts` can be moved into `room.ts` class.
wdyt

Reply:
I thought about it for a while, in the end decided it was a nice encapsulation: https://github.com/ably/ably-chat-js/pull/505#discussion_r2013699894

Reply:
This means we can directly inject a `room` instance into each feature. It also simplifies many things related to DI.

Reply:
It's something we can change in the future if it becomes a problem, I think - I don't see a strong argument for changing things at the moment